### DISCUSSION TITLE
How to get the markup string of rich object instead of printing them on the console?

### DISCUSSION THREAD

**Discussion by 2catycm** [Textualize/rich#3355] [Q&A] [answered]
For example, we may have a RichTree object
```python
root = Tree("🌲 [b green]Rich Tree", highlight=True, hide_root=True)

    node = root.add(":file_folder: Renderables", guide_style="red")
    simple_node = node.add(":file_folder: [bold yellow]Atomic", guide_style="uu green")
    simple_node.add(Group("📄 Syntax", syntax))
    simple_node.add(Group("📄 Markdown", Panel(markdown, border_style="green")))
```
Now, it is certained that we can print the object on the console by `console = Console(); console.print(root)` .
But that's not what I want, I want to find somekind of `to_string` method of Tree, I want to know the markup of colors and other formats of `root`. 
I want to see the str like "[bold green]Processing[/bold green]" with "[bold green]". 
Once I have the string markup, I can do automatic processing on that.

**Comment by willmcgugan** [comment, marked as answer]
https://rich.readthedocs.io/en/latest/console.html#capturing-output

**Reply by 2catycm** [reply]
Thanks for your answer.
